ui/src/components/Sidebar.tsx

```diff
@@ -534,6 +534,168 @@ function ExportJSON() {
+              return pod1.x - pod2.x;
+            } else {
+              return pod1.y - pod2.y;
+            }
+          } else {
+            return 0;
+          }
+        });
+
+      for (let i = 0; i < sortedChildren.length; i++) {
+        let pod = pods[sortedChildren[i]];
+        let geoScore = curScore + `${i + 1}`;
+        adj[pod.id] = {
+          name: pod.name,
+          parentId: pod.parent,
+          geoScore: geoScore,
+        };
+
+        if (pod.type == "SCOPE") {
+          q.push([pod, geoScore]);
+        } else if (pod.type == "CODE") {
+          jupyterCellList.push({
+            cell_type: "code",
+            // hard-code execution_count
+            execution_count: 1,
+            // TODO: expand other Codepod related-metadata fields, or run a real-time search in database when importing.
+            metadata: { id: pod.id, geoScore: Number(geoScore) },
+            source: [pod.content || ""],
+          });
+        } else if (pod.type == "RICH") {
+          jupyterCellList.push({
+            cell_type: "markdown",
+            // hard-code execution_count
+            execution_count: 1,
+            // TODO: expand other Codepod related-metadata fields, or run a real-time search in database when importing.
+            metadata: { id: pod.id, geoScore: Number(geoScore) },
+            source: [pod.richContent || ""],
+          });
+        }
+      }
+    }
+
+    // sort the generated cells by their geoScore
+    jupyterCellList.sort((cell1, cell2) => {
+      if (
+        Number(cell1.metadata["geoScore"]) < Number(cell2.metadata["geoScore"])
+      ) {
+        return -1;
+      } else {
+        return 1;
+      }
+    });
+
+    // Append the scope structure as comment for each cell and format source
+    for (const cell of jupyterCellList) {
+      let scopes: string[] = [];
+      let parentId = adj[cell.metadata["id"]].parentId;
+
+      // iterative {parentId,name} retrieval
+      while (parentId && parentId != "ROOT") {
+        scopes.push(adj[parentId].name);
+        parentId = adj[parentId].parentId;
+      }
+
+      // Add scope structure as a block comment at the head of each cell
+      let scopeStructureAsComment =
+        scopes.length > 0
+          ? [
+              "'''\n",
+              `CodePod Scope structure: ${scopes.reverse().join("/")}\n`,
+              "'''\n",
+            ]
+          : [""];
+
+      const sourceArray = cell.source[0]
+        .split(/\r?\n/)
+        .map((line) => line + "\n");
+
+      cell.source = [...scopeStructureAsComment, ...sourceArray];
+    }
+
+    const fileContent = JSON.stringify({
+      // hard-code Jupyter Notebook top-level metadata
+      metadata: {
+        name: repoName,
+        kernelspec: {
+          name: "python3",
+          display_name: "Python 3",
+        },
+        language_info: { name: "python" },
+        Codepod_version: "v0.0.1",
+      },
+      nbformat: 4,
+      nbformat_minor: 0,
```

Comment:
The latest nbformat version seems to be [5.9.0](https://nbformat.readthedocs.io/en/latest/changelog.html#id2). Should we use that instead?

Reply:
Colab still uses `nbformat:4` in their exported notebook, and it doesn't support the 5.9.0 yet. 

<img width="856" alt="Screen Shot 2023-06-30 at 9 19 24 PM" src="https://github.com/codepod-io/codepod/assets/10226241/7d7c6edd-1e3d-4bcd-90a9-31d28f441d7b">